### PR TITLE
Add a side-by-side layout to the .display configuration

### PR DESCRIPTION
**Pull request by AlanGriffiths**
When creating a .display configuration include multiple layouts (`all_at_00` and `side_by_side`)

### CODE REVIEW

**Review by Saviq — CHANGES_REQUESTED** [review]
I know it's a bit of a chicken'n'egg problem, but if I run `ubuntu-frame --display-layout=side_by_side`, it won't work the first time around. Need to restart to get the config in.

The config dumped into the logs is confusing, because it says `layouts:  default:` and proceeds to list the selected layout, not the default - it also doesn't list the non-selected layouts. I suppose that's a separate problem, though?

File: src/miral/static_display_config.cpp [hunk 1]
```diff
@@ -510,22 +504,43 @@ void miral::ReloadingYamlFileDisplayConfig::dump_config(std::function<void(std::
...
 
-        if (access(filename.c_str(), F_OK))
+        // We don't want to write a .display config if running on an X11 or Wayland platform
+        // (We can not configure those this way). We guess the platform based on the output names
+        // because KMS sets realistic names.
+        bool all_fake_outputs = true;
+        conf.for_each_output([&all_fake_outputs](mg::UserDisplayConfigurationOutput const& conf_output)
+            {
+                all_fake_outputs &= conf_output.name.starts_with("OUT-");
+            });
```
[the remark this answers is not in the document]
Comment (AlanGriffiths) [reply src/miral/static_display_config.cpp:514]: OK, I'll drop this. It should become less of an issue once I get to consolidating "hardware" changes

**Comment by AlanGriffiths** [comment]
> The config dumped into the logs is confusing, because it says `layouts: default:` and proceeds to list the selected layout, not the default - it also doesn't list the non-selected layouts. I suppose that's a separate problem, though?

The config dumped into the log is (still) the layout that has been applied. I thought that would be less confusing than changing it to dump multiple layouts with no indication which has become active.

**Comment by Saviq** [comment]
> The config dumped into the log is (still) the layout that has been applied. I thought that would be less confusing than changing it to dump multiple layouts with no indication which has become active.

Yeah, but if it listed the _actual_ layout used, that would be less confusing.

> The original purpose of this log message was to aid creating the .display configuration file. Maybe it is now obsolete and we should simply stop spamming the log with it?

That's another option, yes. Since this only makes sense when using static config…

**Comment by AlanGriffiths** [comment]
> > The config dumped into the log is (still) the layout that has been applied. I thought that would be less confusing than changing it to dump multiple layouts with no indication which has become active.
> 
> Yeah, but if it listed the _actual_ layout used, that would be less confusing.

Well, the anchor and alias stuff complicates that. Maybe I should drop that and just rename `all_at_00` to `default`?

**Comment by Saviq** [comment]
> Well, the anchor and alias stuff complicates that. Maybe I should drop that and just rename `all_at_00` to `default`?

That may well be my preference, we don't want to teach people about YAML anchors ;)

And `all_at_00` is the current default… we should think of finding a real clone layout, too (I know not easy, what with different preferred modes, aspect ratios…).

**Comment by AlanGriffiths** [comment]
> we should think of finding a real clone layout, too (I know not easy, what with different preferred modes, aspect ratios…).

https://github.com/MirServer/mir/issues/921?

**Review by Saviq — APPROVED** [review]
Sures.

bors merge

**Pull request merged**